File: 0041a-translation/solution.cpp
#include <iostream>
#include <string>

void solution() {
    std::string s, t;
    std::cin >> s >> t;

    bool result = true;
    if (s.size() != t.size()) {
        result = false;
    } else {
        std::string::const_iterator p = s.begin();
        std::string::const_reverse_iterator q = t.rbegin();
        while (p != s.end()) {
            if (*p != *q) {
                result = false;
                break;
            }

            ++p;
            ++q;
        }
    }

    std::cout << (result ? "YES" : "NO") << '\n';
}


void setup() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
}


int main() {
    setup();
    solution();
}

File: 0041a-translation/solution.py
def solution():
    s, t = input(), input()

    result = True
    if len(s) != len(t):
        result = False
    else:
        i, j = 0, len(t) - 1
        while i < len(s):
            if s[i] != t[j]:
                result = False
                break

            i += 1
            j -= 1

    answer = "YES" if result else "NO"
    print(answer)


if __name__ == "__main__":
    solution()
